sqlmesh/core/context.py

```diff
@@ -877,12 +882,14 @@ def plan(
                 environment or c.PROD,
                 snapshots=self._snapshots(models_override),
                 create_from=create_from,
+                force_no_diff=(restate_models is not None and not expanded_restate_models)
```

Comment:
Why do we need this?

Reply:
If you don't do this then you will have a diff since you are including snapshots and other configuration in plan.